File: solutions/409.longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        vector<int> count(128);
        int odds = 0;
        for(char c : s) 
            odds += ++count[c] & 1 ? 1 : -1;
        return s.size() - odds + (odds > 0);
    }
};

File: solutions/409.longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Set<Character> set = new HashSet<>();
        int res = 0;
        for(char c : s.toCharArray()) {
            if(set.remove(c)) res += 2;
            else set.add(c);
        }
        if(set.isEmpty()) return res;
        return res + 1;
    }
}